src/selectors/walletMigrationArchanova.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2021 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import { createSelector } from 'reselect';
+
+// Constants
+import { ARCHANOVA_WALLET_ASSET_MIGRATION } from 'constants/archanovaConstants';
+import { TX_PENDING_STATUS } from 'constants/historyConstants';
+
+// Selectors
+import { ethereumRatesSelector } from 'selectors';
+import { archanovaWalletAssetsBalancesSelector, archanovaCollectiblesSelector } from 'selectors/archanova';
+import { archanovaAccountEthereumHistorySelector } from 'selectors/history';
+
+// Utils
+import { mapNotNil } from 'utils/array';
+import { findCollectible } from 'utils/collectibles';
+import { hasNonNegligileWalletBalances } from 'utils/walletMigrationArchanova';
+
+// Types
+import type { RootReducerState, Selector } from 'reducers/rootReducer';
+import type { WalletAssetsBalances } from 'models/Balances';
+import type { CollectibleId, Collectible } from 'models/Collectible';
+import type { RatesByAssetAddress } from 'models/Rates';
+import type { Transaction } from 'models/Transaction';
+
+
+/**
+ * Returns archanova account collectibles selected for migration.
+ */
+export const collectiblesToMigrateSelector: Selector<Collectible[]> = createSelector(
+  archanovaCollectiblesSelector,
+  (root: RootReducerState) => root.walletMigrationArchanova.collectiblesToMigrate,
+  (archanovaCollectibles: Collectible[], collectiblesToMigrate: CollectibleId[]): Collectible[] => {
+    return mapNotNil(collectiblesToMigrate, (id) => findCollectible(archanovaCollectibles, id));
+  },
+);
+
+// Checks for non-negligible balances
+export const showWalletMigrationArchanovaSelector: Selector<boolean> = createSelector(
+  archanovaWalletAssetsBalancesSelector,
+  archanovaCollectiblesSelector,
+  ethereumRatesSelector,
+  (walletBalances: WalletAssetsBalances, collectibles: Collectible[], rates: RatesByAssetAddress): boolean => {
+    return hasNonNegligileWalletBalances(walletBalances, rates) || !!collectibles.length;
```

Comment:
Do you think we can replace `hasNonNegligileWalletBalances` into something easier to type and read like `hasEnoughWalletBalancesToMigrate`?

Reply:
"Enough" would suggest we are talking about having enough ETH to pay for the tx fee. Here we are checking that user funds are potentially worth migrating (i.e. are non-negligible). Feel free to suggest another name that more succinctly describes that concept.

Reply:
Not necessarily, we don't say anything about ETH and tx so it would have no relations to that, however, if my suggestion still confuses you and I cannot come with any other options we can keep, but I think there's a typo on `negligile` which should be `negligible`.